### PR TITLE
feat: add max reponse body limit

### PR DESCRIPTION
**Pull request by trim21**
Client not will be able to check resposne body size limit.

If server return a large response than expected, it will discard the response and return a err `ErrResponseBodyTooLarge`. 

```golang
c := Client().SetResponseBodyLimit(1024*1024*1024) // do not allow body larger than 1GiB
res, err := c.R().Get(...)
if err != nil{
  if errors.Is(err, resty.ErrResponseBodyTooLarge){
    ...
  }
}
```

close https://github.com/go-resty/resty/issues/819

### CODE REVIEW

**Comment by jeevatkm** [comment]
@trim21 Thanks for creating a PR.

I wanted to confirm before I start reviewing it. If you don't mind, your use case is to fail the request if the response body size exceeds the specific size. Can you update the PR description to one use case and its example?

**Comment by trim21** [comment]
> @trim21 Thanks for creating a PR.
> 
> I wanted to confirm before I start reviewing it. If you don't mind, your use case is to fail the request if the response body size exceeds the specific size. Can you update the PR description to one use case and its example?

is this ok?

File: client.go [hunk 1]
```diff
@@ -1089,6 +1090,18 @@ func (c *Client) SetJSONEscapeHTML(b bool) *Client {
...
 }
 
+// SetResponseBodyLimit set a max body size limit on response, avoid reading too many data to memory.
+//
+// Client will return [ErrResponseBodyTooLarge] if uncompressed response body size if larger than limit.
+// Body size limit will not be enforced in following case:
+// - ResponseBodyLimit is 0, which is the default behavior.
+// - [Response.SetOutput] is called to save a response data to file.
+// - `DoNotParseResponse` is set for client or request.
+func (c *Client) SetResponseBodyLimit(v int64) *Client {
```
Comment (jeevatkm) [inline client.go:1100]: @trim21 Please test the godoc display by running godoc locally, e.g.:  bullet points in the godoc.

[the remark this answers is not in the document]
Comment (trim21) [reply client.go:1100]: I don't think this is necessary, https://pkg.go.dev/ already support this:

image

[the remark this answers is not in the document]
Comment (trim21) [reply client.go:1100]: image

**Comment by trim21** [comment]
I'm not sure size limit should be `int` or `int64`.

It's a length limit of `[]byte`, and `len([]byte)` is int, so it would make more sense to be int?

**Comment by jeevatkm** [comment]
> I'm not sure size limit should be `int` or `int64`.
> 
> It's a length limit of `[]byte`, and `len([]byte)` is int, so it would make more sense to be int?

@trim21 Yes, for size limit, keep type `int`

File: client.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeevatkm) [reply client.go]: Okay

**Review by jeevatkm — DISMISSED** [review]
@trim21 Thanks for updating the PR.

**Comment by jeevatkm** [comment]
@trim21 Can you please check this https://github.com/go-resty/resty/pull/830/checks?check_run_id=29386261232 and improve it?

**Comment by trim21** [comment]
> @trim21 Can you please check this https://github.com/go-resty/resty/pull/830/checks?check_run_id=29386261232 and improve it?

Wow can I make test client read a error from response?

**Review by jeevatkm — APPROVED** [review]
@trim21 Thanks for the test coverage updates. Now all green.

**Pull request merged**